src/os/bluestore/BlueStore.h

```diff
@@ -688,7 +715,10 @@ class BlueStore : public ObjectStore,
       if (--nref == 0)
 	delete this;
     }
+
+#ifdef WITH_ESB
```

Comment:
nit: we might wrap ~Blob's implementation with ifdef while leaving its declaration for both ESB/non-ESB intact. Just to have fewer ifdefs which generally speaking I don't like...